templates/vault.systemd.erb

```diff
@@ -20,20 +23,25 @@ ProtectHome=read-only
 <% # Still require check for :undef for Puppet 3.x -%>
 <% if scope['vault::disable_mlock'] && scope['vault::disable_mlock'] != :undef -%>
 CapabilityBoundingSet=CAP_SYSLOG
-NoNewPrivileges=yes
 <% else -%>
 SecureBits=keep-caps
 Capabilities=CAP_IPC_LOCK+ep
 CapabilityBoundingSet=CAP_SYSLOG CAP_IPC_LOCK
-NoNewPrivileges=yes
 <% end -%>
+NoNewPrivileges=yes
 Environment=GOMAXPROCS=<%= scope['vault::num_procs'] %>
 ExecStart=<%= scope['vault::bin_dir'] %>/vault server -config=<%= scope['vault::config_dir'] %>/config.json <%= scope['vault::service_options'] %>
 KillSignal=SIGINT
 TimeoutStopSec=30s
 Restart=on-failure
 StartLimitInterval=60s
 StartLimitBurst=3
+AmbientCapabilities=CAP_IPC_LOCK
+ExecReload=/bin/kill --signal HUP $MAINPID
+KillMode=process
+RestartSec=5
+TimeoutStopSec=30
+LimitNOFILE=65536
```

Comment:
Personally, I don't think that such a high value makes sense. This high limit is only required if vault receives a huge amount of requests. Such a high nomber of open file(descriptors) in a small environment indicates a bug and this systemd setting will hide it.